demonstrations/tutorial_qgrnn.py

```diff
@@ -0,0 +1,609 @@
+
+    # Applies a layer of coupling gates (based on a graph)
+    for count, i in enumerate(graph.edges):
+        qml.MultiRZ(2 * param1[count] * trotter, wires=[i[0], i[1]])
+
+    # Applies a layer of RZ gates
+    for count, i in enumerate(qubits):
+        qml.RZ(2 * param2[count] * trotter, wires=i)
+
+    # Applies a layer of RX gates
+    for i in qubits:
+        qml.RX(2 * trotter, wires=i)
+
+
+######################################################################
+# As was mentioned in the first section, the QGRNN has two qubit
+# registers. In one register, some piece of quantum data
+# :math:`|\psi(t)\rangle` is prepared and in the other we have
+# :math:`U_{H}(\boldsymbol\mu, \ \Delta) |\psi_0\rangle`. We need a
+# way to measure the similarity between these states. 
+# One way that we can do this is by using the fidelity, which is
+# simply the modulus squared of the inner product between the states,
+# :math:`| \langle \psi(t) | U_{H}(\Delta, \ \boldsymbol\mu) |\psi_0\rangle |^2`.
+# To calculate this value, we utilize a `SWAP
+# test <https://en.wikipedia.org/wiki/Swap_test>`__ between the registers:
+#
+
+
+def swap_test(control, register1, register2):
+
+    qml.Hadamard(wires=control)
+    for i in range(0, len(register1)):
+        qml.CSWAP(wires=[int(control), register1[i], register2[i]])
+    qml.Hadamard(wires=control)
+
+
+######################################################################
+# Before creating the full QGRNN and the cost function, we
+# define a few more fixed values. Among these fixed values is a "guessed"
+# interaction graph, which we define to be the complete graph. This choice 
+# is motivated by the fact that any target interaction graph will be a subgraph 
+# of this initial guess. 
+#
+
+# Defines some fixed values
+
+reg1 = list(range(qubit_number))  # First qubit register
+reg2 = list(range(qubit_number, 2 * qubit_number))  # Second qubit register
+
+control = 2 * qubit_number  # Index of control qubit
+trotter_step = 0.01  # Trotter step size
+
+# Defines the interaction graph for the new qubit system
+
+new_ising_graph = nx.Graph()
+new_ising_graph.add_nodes_from(range(qubit_number, 2 * qubit_number))
+new_ising_graph.add_edges_from([(4, 5), (5, 6), (6, 7), (4, 6), (7, 4), (5, 7)])
+
+print("Edges: " + str(new_ising_graph.edges))
+nx.draw(new_ising_graph)
+
+
+######################################################################
+# Part of the idea behind the QGRNN is that
+# we don’t know the interaction graph, and it has to be learned. In this case, the graph  
+# is learned **automatically** as the target parameters are learned. The
+# :math:`\boldsymbol\mu` parameters that correspond to edges that don't exist in 
+# the target graph will simply approach :math:`0`.
+#
+# With this done, we implement the QGRNN circuit for some given time value:
+#
+
+
+def qgrnn(params1, params2, time=None):
+
+    # Prepares the low energy state in the two qubit registers
```

Comment:
```suggestion
    # Prepares the low energy state in the two registers
```
Across the tutorial: it may be better to just use register as opposed to qubit register